### ISSUE TITLE
incorrect result of #![deny(unused_qualifications)] in match block

### ISSUE THREAD

**Issue by eastfisher** [rust-lang/rust-clippy#9372]
### Summary

#![deny(unused_qualifications)] does not work in match block.

### Lint Name

#![deny(unused_qualifications)]

### Reproducer

I tried this code:

tree:

|- main.rs
|- status.rs

```rust
// main.rs

#![deny(unused_qualifications)]
#![allow(dead_code)]

mod status;

use status::StatusCode;

fn main() -> anyhow::Result<()> {
    let st = StatusCode::Ok;
    let resp = status::Resp{status: st};
    match resp.status() {
        status::StatusCode::Ok => {Ok(())} // lint should warn but not
        status::StatusCode::NotFound => {Ok(())}  // lint should warn but not
    }
}
```

```rust
// status.rs

#[derive(Clone)]
pub enum StatusCode {
    Ok,
    NotFound,
}

pub struct Resp {
    pub(crate) status: StatusCode,
}

impl Resp {
    pub fn status(&self) -> StatusCode {
        self.status.clone()
    }
}
```

run the following cmd:

```bash
cargo clippy
```

I expected to see this happen:

> error: unnecessary qualification

Instead, this happened:

> no error

### Version

```text
rustc 1.62.1 (e092d0b6b 2022-07-16)
binary: rustc
commit-hash: e092d0b6b43f2de967af0887873151bb1c0b18d3
commit-date: 2022-07-16
host: x86_64-apple-darwin
release: 1.62.1
LLVM version: 14.0.5
```

**Comment by Xuanwo** [comment]
I can't find `unused-qualifications` in https://rust-lang.github.io/rust-clippy/master/index.html.

I'm guessing it's a `rustc` lint?

**Comment by giraffate** [comment]
The `unused_qualifications` is the rustc's lint: https://doc.rust-lang.org/stable/nightly-rustc/rustc_lint/builtin/static.UNUSED_QUALIFICATIONS.html, so it would be better that this issue is moved to [rust-lang/rust](https://github.com/rust-lang/rust).

The `main.rs` bring `status::StatusCode` in that scope, but it doesn't use `status::StatusCode` as it is. It uses `status::StatusCode::Ok` and `status::StatusCode::NotFound`, so the `unused_qualifications` lint doesn't fire.

**Comment by eastfisher** [comment]
@giraffate Thanks for answer! I start another [issue](https://github.com/rust-lang/rust/issues/100979) in rust-lang/rust.
